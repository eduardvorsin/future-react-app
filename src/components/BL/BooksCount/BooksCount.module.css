.books-count {
	margin: 0;
	font-weight: 700;
	text-align: center;
	margin-bottom: 20px;
	font-size: 24px;
	color: var(--text-color-primary);
}