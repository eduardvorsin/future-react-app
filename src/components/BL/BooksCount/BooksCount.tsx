import React from 'react';
import classes from './BooksCount.module.css';

type BooksCountProps = {
  className?: string,
  children: React.ReactNode,
}

export const BooksCount: React.FC<BooksCountProps> = ({
  className,
  children,
}) => {
  const booksCountClasses = [
    className,
    classes['books-count'],
  ].join(' ');

  return (
    <p
      className={booksCountClasses}
    >
      {children}
    </p>
  );
};
export default BooksCount;
