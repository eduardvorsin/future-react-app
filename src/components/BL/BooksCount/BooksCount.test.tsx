import React from 'react';
import { render, screen } from '@testing-library/react';
import BooksCount from './BooksCount';
import i18n from '../../../localization/i18next';

describe('BooksCount tests', () => {
  it('is rendered correctly', () => {
    render(
      <BooksCount
        itemsCount={0}
        totalItemsCount={1}
      />,
    );

    expect(screen.getByText<HTMLParagraphElement>(/book/i)).toBeInTheDocument();
  });

  it('is a basic snapshot', () => {
    render(
      <BooksCount
        itemsCount={0}
        totalItemsCount={1}
      />,
    );

    expect(screen.getByText<HTMLParagraphElement>(/book/i)).toMatchSnapshot();
  });

  it('is a snapshot when 0 books', () => {
    render(
      <BooksCount
        itemsCount={0}
        totalItemsCount={0}
      />,
    );

    expect(screen.getByText<HTMLParagraphElement>(/book/i)).toMatchSnapshot();
  });
});

describe('BooksCount integration tests', () => {
  it('the component with the text in English is displayed correctly', () => {
    const localizedText = i18n.t('booksCount', {
      count: 22,
      currentBooksCount: 11,
      ns: 'homePage',
    });

    render(
      <BooksCount
        itemsCount={11}
        totalItemsCount={22}
      />,
    );

    expect(screen.getByText<HTMLParagraphElement>(localizedText)).toBeInTheDocument();
  });

  it('the component with the text in Russian is displayed correctly', () => {
    i18n.changeLanguage('ru');
    const localizedText = i18n.t('booksCount', {
      count: 22,
      currentBooksCount: 11,
      ns: 'homePage',
    });

    render(
      <BooksCount
        itemsCount={11}
        totalItemsCount={22}
      />,
    );

    expect(screen.getByText<HTMLParagraphElement>(localizedText)).toBeInTheDocument();
  });
});
