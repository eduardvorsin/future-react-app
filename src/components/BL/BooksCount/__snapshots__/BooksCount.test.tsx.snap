// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BooksCount tests is a basic snapshot 1`] = `
<p
  class="0 books-count"
>
  Presented 0 of (1) book
</p>
`;

exports[`BooksCount tests is a snapshot when 0 books 1`] = `
<p
  class="0 books-count"
>
  No suitable books have been found for your request
</p>
`;
